curriculum/challenges/english/25-front-end-development/lab-factorial-calculator/66c07238b01053abaf812065.md

```diff
@@ -0,0 +1,99 @@
+---
+id: 66c07238b01053abaf812065
+title: Build a Factorial Calculator
+challengeType: 14
+dashedName: build-a-factorial-calculator
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+A factorial is the product of an integer and all the integers below it. For example, the factorial of `5` is `5 * 4 * 3 * 2 * 1 = 120`. In this lab, you will create a factorial calculator that takes a number from the user and calculates the factorial of that number.
+
+1. You should declare a variable `num` and assign it the value of `5`.
```

Comment:
I would ask to assign a generic number here

Reply:
After discussion in the chat room, we decided to use a predefined number here.

Reply:
One of the reasons why I suggested using 5 is because that is a reasonable number for a factorial.

What we don't want is for the user to pick 1,000,000 or something like that.

If we want users to pick their own number we do need to put limits. So I vote the limit be 20. That seems reason for this exercise.

Then we would just need to test for the following:
- `num` should actually be a number and not some sort of string, or anything else
- the number needs to be a positive number between 1 -20

those should be easy to test with built in chai assert methods